File: LeetCode575.java
import java.util.*;

class Solution {
    public int distributeCandies(int[] candyType) {
        Set<Integer> set = new HashSet<>();
        int size = candyType.length;
        for (int i = 0; i < size; i++) {
            set.add(candyType[i]);
        }
        return set.size() == size / 2 ? size / 2 : set.size();
    }
}

File: leetcode575.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int distributeCandies(vector<int> &candyType)
    {
        unordered_set<int> uset;
        for (auto element : candyType)
        {
            uset.insert(element);
        }
        return uset.size() >= candyType.size() / 2 ? candyType.size() / 2 : uset.size();
    }
};